src/utils/hooks/useLoadApplicationNew.ts

```diff
@@ -0,0 +1,201 @@
+
+  const { data, error } = useGetApplicationNewQuery({
+    variables: {
+      serial: serialNumber,
+    },
+    // pollInterval: 500,
+    // skip: isLoading,
+    fetchPolicy: networkFetch ? 'network-only' : 'cache-first',
+  })
+
+  // const { error: triggerError, isTriggerProcessing } = useTriggerProcessing({
+  //   isApplicationLoaded: data !== undefined,
+  //   serialNumber,
+  //   // triggerType: 'applicationTrigger',
+  // })
+
+  // const { data: statusData, error: statusError } = useGetApplicationStatusQuery({
+  //   variables: { serial: serialNumber },
+  //   skip: isTriggerProcessing,
+  //   fetchPolicy: 'network-only',
+  // })
+
+  useEffect(() => {
+    // if (!data || !statusData || isTriggerProcessing) return
+    if (!data) return
+    const application = data.applicationBySerial as Application
+
+    if (!application) {
+      setStrucutreError('No application found')
+    }
+
+    const { id, code, name, startMessage } = application.template as Template
+
+    setTemplate({
+      id,
+      code,
+      name: name as string,
+      startMessage: startMessage ? startMessage : undefined,
+    })
+
+    const applicationSection = application.applicationSections.nodes as ApplicationSection[]
+    const sections = getApplicationSections(applicationSection)
+
+    const templateStages = application.template?.templateStages.nodes as TemplateStage[]
+
+    const stagesDetails: ApplicationStages = {
+      stages: templateStages.map((stage) => ({
+        number: stage.number as number,
+        title: stage.title as string,
+        description: stage.description ? stage.description : undefined,
+      })),
+      submissionMessage: application.template?.submissionMessage as string,
+    }
+
+    const applicationDetails: ApplicationDetails = {
+      id: application.id,
+      type: application.template?.name as string,
+      isLinear: application.template?.isLinear as boolean,
+      serial: application.serial as string,
+      name: application.name as string,
+      outcome: application.outcome as string,
+    }
+
+    const templateElements = [] as TemplateElementState[]
+    application.applicationSections.nodes.forEach((sectionNode) => {
+      let count = 1
+      const elementsInSection = sectionNode?.templateSection?.templateElementsBySectionId
+        ?.nodes as TemplateElement[]
+      elementsInSection.forEach((element) => {
+        if (element.elementTypePluginCode === 'pageBreak') count++
+        else
+          templateElements.push({
+            ...element,
+            pluginCode: element.elementTypePluginCode,
+            sectionIndex: sectionNode?.templateSection?.index,
+            sectionCode: sectionNode?.templateSection?.code,
+            elementIndex: element.index,
+            page: count,
+          } as TemplateElementState)
+      })
+    })
+
+    evaluateElementExpressions({ currentUser: currentUser as User, templateElements }).then(
```

Comment:
isVisible, isRequired and is, isEditable requires applications response by code. The plan was to get them via useGetFullApplicationStructure

Reply:
So that means that the evaluation will have to wait until that hook, not this one?

Reply:
Agreed. I'll be changing the type to have those fields as optional ones.

Reply:
Done: 0b61d1c